doc/shared/sundomeigest/SUNDomEigEst_API.rst

```diff
@@ -0,0 +1,622 @@
+   Copyright (c) 2002-2025, Lawrence Livermore National Security
+   and Southern Methodist University.
+   All rights reserved.
+
+   See the top-level LICENSE and NOTICE files for details.
+
+   SPDX-License-Identifier: BSD-3-Clause
+   SUNDIALS Copyright End
+   ----------------------------------------------------------------
+
+.. _SUNDomEigEst.API:
+
+The SUNDomEigEstimator API
+=============================
+
+The SUNDomEigEst API defines several dominant eigenvalue estimation operations that enable
+SUNDIALS packages to utilize this API.  These functions can be divided into three categories.
+The first are the core dominant eigenvalue estimation functions.  The second consist of "set"
+routines to supply the dominant eigenvalue estimator with functions provided by the SUNDIALS
+packages and to modify estimator parameters.  The final group consists of "get" routines for
+retrieving dominant eigenvalue estimation statistics.  All of these functions are defined in
+the header file ``sundials/sundials_domeigestimator.h``.
+
+
+.. _SUNDomEigEst.CoreFn:
+
+SUNDomEigEstimator core functions
+-----------------------------------------------------
+
+The core dominant eigenvalue estimator functions consist of several **required**
+functions: :c:func:`SUNDomEigEst_NewEmpty` creates a new empty ``SUNDomEigEstimator`` 
+object, :c:func:`SUNDomEigEst_SetATimes` provides a :c:type:`SUNATimesFn` function pointer,
+as well as a ``void*`` pointer to a data structure used by this routine,
+:c:func:`SUNDomEigEst_Initialize` initializes the estimator object once all estimator-specific
+options have been set, :c:func:`SUNDomEig_Estimate` estimates the dominant eigenvalue,
+:c:func:`SUNDomEigEst_FreeEmpty` an empty estimator and :c:func:`SUNDomEigEst_Destroy` destroys 
+an estimator object.
+
+.. c:function:: SUNDomEigEstimator SUNDomEigEst_NewEmpty(SUNContext sunctx)
+
+   *Required.*
+
+   This function allocates a new ``SUNDomEigEstimator`` object and
+   initializes its content pointer and the function pointers in the operations
+   structure to ``NULL``.
+
+   **Arguments:**
+
+      * *sunctx* -- the :c:type:`SUNContext` object (see :numref:`SUNDIALS.SUNContext`)
+
+   **Return value:**
+
+      If successful, this function returns a ``SUNDomEigEstimator`` object.
+      If an error occurs when allocating the object, then this routine will
+      return ``NULL``.
+
+
+.. c:function:: SUNErrCode SUNDomEigEst_SetATimes(SUNDomEigEstimator DEE, void* A_data, SUNATimesFn ATimes)
+
+   *Required.*
+
+   Provides a :c:type:`SUNATimesFn` function pointer, as well as a ``void*`` pointer to a
+   data structure used by this routine, to the dominant eigenvalue estimator object
+   *DEE*.  SUNDIALS packages call this function to set the matrix-vector product function
+   to either an estimator-provided difference-quotient via vector operations or a user-supplied
+   estimator-specific routine.
+
+   **Arguments:**
+
+      * *DEE* -- a SUNDomEigEst object,
```

Comment:
Since the object name is "SUNDomEigEstimator", but the function prefix is "SUNDomEigEst", then I recommend that these function descriptions use the correct object name instead of the shortened function prefix: 
```suggestion
      * *DEE* -- a SUNDomEigEstimator object,
```